consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/validation/ProposalPayloadValidator.java

```diff
@@ -89,4 +104,58 @@ private boolean validateBlock(final Block block) {
 
     return true;
   }
+
+  /*
+    Validate CMS in block header (only for PKI-mode)
+  */
+  private boolean validateCms(final Block block) {
+    final PkiQbftContext pkiQbftContext;
+    try {
+      pkiQbftContext = protocolContext.getConsensusState(PkiQbftContext.class);
+    } catch (final ClassCastException e) {
+      // TODO-lucas Do we have a better way of doing this?
+      // Not running on PKI-enabled mode
+      return true;
+    }
+
+    final BftExtraData extraData =
+        pkiQbftContext.getBlockInterface().getExtraData(block.getHeader());
+    final PkiQbftExtraData pkiExtraData = (PkiQbftExtraData) extraData;
+    final KeyStoreWrapper trustStore =
+        pkiQbftContext.getPkiBlockCreationConfiguration().getTrustStore();
+
+    if (pkiExtraData.getCms().isPresent()) {
+      // TODO-lucas Do we need to do this or could I just use 'block.getHeader().getHash()'
+      final Hash proposedBlockHash =
+          BftBlockHeaderFunctions.forOnChainBlock(new PkiQbftExtraDataCodec())
+              .hash(block.getHeader());
+
+      LOG.info(">>> Validating CMS for block {}", proposedBlockHash);
+
+      CertStore crlCertStore = null;
+      if (trustStore.getCRLs() != null) {
+        try {
+          crlCertStore =
+              CertStore.getInstance(
+                  "Collection", new CollectionCertStoreParameters(trustStore.getCRLs()));
+        } catch (final Exception e) {
+          throw new RuntimeException("Error reading PKI Block Creation TrustStore CRL file", e);
+        }
+      }
+
+      final CmsValidator cmsValidator = new CmsValidator(trustStore, crlCertStore);
+
+      if (!cmsValidator.validate(pkiExtraData.getCms().get(), proposedBlockHash)) {
+        LOG.info(">>> CMS Validation INVALID for block {}", proposedBlockHash);
+        return false;
+      } else {
+        LOG.info(">>> CMS Validation VALID for block {}", proposedBlockHash);
+        return true;
+      }
+    } else {
```

Comment:
return false in this case? Would have though that all proposals must have a CMS if PKI mode is being used